### ISSUE TITLE
AMAI installer incluse all features of AMAIStrategyManager.pl

### ISSUE THREAD

**Issue by paulo101977** [SMUnlimited/AMAI#93]
I am opening this issue to remember to implement the remaining features of AMAIStrategyManager.pl. I just don't know if I can remove perl completely as there is a script that is complex and difficult to port to javascript.

**Comment by SMUnlimited** [comment]
Yeah that is difficult as its basically its own version of JASS that uses Perl to compile down to normal JASS. 

So I wouldn't worry about that and instead just replicate the bat scripts, unless you want to rewrite most of AMAI  😛

**Comment by paulo101977** [comment]
> Yeah that is difficult as its basically its own version of JASS that uses Perl to compile down to normal JASS.
> 
> So I wouldn't worry about that and instead just replicate the bat scripts, unless you want to rewrite most of AMAI 😛

Some things I want to refactor and do with js, even to be in the same language as electron. However, these others in perl, I will migrate little by little. If possible, of course.

**Comment by jzy-chitong56** [comment]
> Some things I want to refactor and do with js, even to be in the same language as electron. However, these others in perl, I will migrate little by little. If possible, of course.

In China, a player said that it can be transplanted to Lua, but I don't understand this language... If I use Lua, I can't understand it at all

**Comment by paulo101977** [comment]
> In China, a player said that it can be transplanted to Lua, but I don't understand this language... If use Lua, I can't understand it at all

If there is a way to convert lua to javascript it would be worth it. But I would have to research this conversion, if it's possible. I tried converting perl to javascript but it's too complex. It would take a lot of time trying to understand the whole algorithm and do the conversion without any kind of bug.
